control-plane/api-gateway/binding/validation.go

```diff
@@ -644,6 +689,35 @@ func routeKindIsAllowedForListenerExplicit(allowedRoutes *gwv1alpha2.AllowedRout
 	return routeKindIsAllowedForListener(allowedRoutes.Kinds, gk)
 }
 
+func validateAuthFilters(authFilters []*v1alpha1.RouteAuthFilter, resources *common.ResourceMap) authFilterValidationResults {
+	results := make(authFilterValidationResults, 0, len(authFilters))
+
+	for _, filter := range authFilters {
+		if filter == nil {
+			continue
+		}
+		var result authFilterValidationResult
+		missingJWTProviders := make([]string, 0)
+		for _, provider := range filter.Spec.JWT.Providers {
+			if _, ok := resources.GetJWTProviderForGatewayJWTProvider(provider); !ok {
+				missingJWTProviders = append(missingJWTProviders, provider.Name)
```

Comment:
Does it make sense to include the namespace here in case the user has providers with the same name in different namespaces?

Reply:
jwt providers are partition scoped so namespace shouldn't matter here